### ISSUE TITLE
Extensions auto discovery not working in some cases

### ISSUE THREAD

**Issue by bjrmatos** [jsreport/jsreport-core#2]
i was testing lastest jsreport (0.10) and found some cases where auto-discovery is not working..

1.- not working with [jsreport installation instructions](https://github.com/jsreport/jsreport#installation), so i have created a directory named `jsreport-playground`, then `npm install jsreport --save` , `node node_modules/jsreport --init`, `node server.js`, results -> Found 0 extensions (i haved added some `console.log` to see what is happening)

![1](https://cloud.githubusercontent.com/assets/4262050/11521135/a077dfde-9877-11e5-89a8-2d254b0cac1b.gif)

the bug is [caused by this line](https://github.com/jsreport/jsreport-core/blob/master/lib/util/util.js#L54), and can be solved by changing the condition to:

```js
// to ignore cycles in any extension that have a dependency in jsreport or jsreport-core 
if (S(rootPath).contains('node_modules') && (S(file).endsWith('node_modules' + path.sep + 'jsreport') || S(file).endsWith('node_modules' + path.sep + 'jsreport-core'))) {
```

the solution is tested in npm 2 not sure if it will work in npm 3

2.- in the same app (`jsreport-playground`), currently if i need to install an additional extension to the app, jsreport doesn't recognize the extension listed in my app, only recognize extensions inside its node_modules (`jsreport-playground/node_modules/jsreport/node_modules`).. is the right behavior?

i think jsreport should recognize the extensions inside the application dependencies, if i'm not wrong jsreport was doing that in previous versions, must be a bug introduced by the split of jsreport, jsreport-core..

i think we should test very carefully if the auto-discovery feature (with jsreport-core and jsreport) works in npm 2 and 3 (since npm3 change the way to install node modules) , i only have tested in npm 2..

**Comment by pofider** [comment]
Ouch, I will check it on npm 2 and probably  release hotfix 0.10.1 for this in the evening. Thank you.

**Comment by pofider** [comment]
1. You were right. That condition was filtering out extensions. I have reimplemented the autodiscovery now in the way it breadth first search for jsreport.config.json. This means it will apply the extensions at the higher position in the node_modules over the extensions referenced somewhere deep. It works both on npm 3 and 2

2. Yes this is bug for npm 2. I have fixed this now.

I am preparing hotfix 0.10.1 with required changes. It should be live in a hour.
